airbyte-ci/connectors/metadata_service/lib/metadata_service/registry_entry.py

```diff
@@ -0,0 +1,517 @@
+        raise
+    logger.info("Parsed YAML file.")
+    return file_dict
+
+
+def _get_and_parse_json_file(file_path: pathlib.Path) -> dict:
+    """Get and parse the json file.
+
+    Args:
+        file_path (pathlib.Path): The path to the file.
+
+    Returns:
+        dict: The file dictionary.
+    """
+    try:
+        logger.debug(f"Getting and parsing JSON file: `{file_path}`")
+        with open(file_path, "r") as f:
+            file_dict = json.load(f)
+    except Exception as e:
+        logger.exception(f"Error parsing JSON file: `{file_path}`")
+        raise
+    logger.info("Parsed JSON file.")
+    return file_dict
+
+
+@sentry_sdk.trace
+def _get_icon_blob_from_gcs(bucket_name: str, metadata_entry: dict) -> storage.Blob:
+    """Get the icon blob from the GCS bucket.
+
+    Args:
+        bucket (storage.Bucket): The GCS bucket.
+        metadata_entry (dict): The metadata entry.
+
+    Returns:
+        storage.Blob: The icon blob.
+    """
+    connector_docker_repository = metadata_entry["dockerRepository"]
+    icon_file_path = f"{METADATA_FOLDER}/{connector_docker_repository}/latest/{ICON_FILE_NAME}"
+    try:
+        logger.info(f"Getting icon blob for {connector_docker_repository}")
+        gcs_client = get_gcs_storage_client(gcs_creds=os.environ.get("GCS_CREDENTIALS"))
+        bucket = gcs_client.bucket(bucket_name)
+        icon_blob = bucket.blob(icon_file_path)
+        if not icon_blob.exists():
+            raise ValueError(f"Icon file not found for `{connector_docker_repository}`")
+    except Exception as e:
+        logger.exception(f"Error getting icon blob")
+        raise
+    return icon_blob
+
+
+def _get_public_url_for_gcs_file(bucket_name: str, file_path: str, cdn_url: Optional[str] = None) -> str:
+    """Get the public URL to a file in the GCS bucket.
+
+    Args:
+        bucket_name: The name of the GCS bucket.
+        file_path: The path to the file in the bucket.
+        cdn_url: The base URL of the CDN that serves the bucket.
+
+    Returns:
+        The public URL to the file.
+    """
+    return f"{cdn_url}/{file_path}" if cdn_url else f"{DEFAULT_ASSET_URL}/{bucket_name}/{file_path}"
+
+
+def _get_connector_type_from_registry_entry(registry_entry: dict) -> TaggedRegistryEntry:
+    """Get the connector type from the registry entry.
+
+    Args:
+        registry_entry (dict): The registry entry.
+
+    Returns:
+        TaggedRegistryEntry: The connector type and model.
+    """
+    if registry_entry.get(ConnectorTypePrimaryKey.SOURCE.value):
+        return (ConnectorTypes.SOURCE, ConnectorRegistrySourceDefinition)
+    elif registry_entry.get(ConnectorTypePrimaryKey.DESTINATION.value):
+        return (ConnectorTypes.DESTINATION, ConnectorRegistryDestinationDefinition)
+    else:
+        raise Exception("Could not determine connector type from registry entry")
+
+
+def _get_registry_entry_blob_paths(metadata_dict: dict, registry_type: str, pre_release_tag: str | None) -> List[str]:
```

Comment:
I think this is the right place to also upload `latest`? (... I think that's a thing we're supposed to do?)